skimage/data/generate_shapes.py

```diff
@@ -0,0 +1,320 @@
+import numpy as np
+
+from .. import draw
+from .._shared.utils import warn
+
+import collections
+
+Label = collections.namedtuple('Label', 'category, x1, x2, y1, y2')
+Point = collections.namedtuple('Point', 'row, column')
+ShapeProperties = collections.namedtuple('ShapeProperties',
+                                         'min_size, max_size, color')
+ImageShape = collections.namedtuple('ImageShape', 'nrows, ncols, depth')
+
+
+def _generate_rectangle_mask(point, image, shape, random):
+    """Generate a mask for a filled rectangle shape.
+
+    The height and width of the rectangle are generated randomly.
+
+    Parameters
+    ----------
+    point : Point
+        The row and column of the top left corner of the rectangle.
+    image : ImageShape
+        The size of the image into which this shape will be fit.
+    shape : ShapeProperties
+        The minimum and maximum size and color of the shape to fit.
+    random : np.random.RandomState
+        The random state to use for random sampling.
+
+    Raises
+    -----
+    ArithmeticError
+        When a shape cannot be fit into the image with the given starting
+        coordinates (x_0, y_0). This usually means the image dimensions are too
+        small or shape dimensions too large.
+
+    Returns
+    -------
+    mask : 3-D array
+        The shape mask that can be applied to an image.
+    label : Label
+        A tuple specifying the category of the shape, as well as its x1, x2, y1
+        and y2 bounding box coordinates.
+    """
+    available_width = min(image.ncols - point.column, shape.max_size)
+    if available_width < shape.min_size:
+        raise ArithmeticError('cannot fit shape to image')
+    available_height = min(image.nrows - point.row, shape.max_size)
+    if available_height < shape.min_size:
+        raise ArithmeticError('cannot fit shape to image')
+    # Pick random widths and heights.
+    r, c = random.randint(shape.min_size, available_width + 1, size=2)
+    mask = np.zeros((image.nrows, image.ncols, image.depth), dtype=np.uint8)
+    mask[point.row:point.row + r, point.column:point.column + c] = shape.color
+    assert mask.sum() > 0
```

Comment:
It used to be a check that I was calculating the dimensions correctly (at edge cases), but now no longer makes sense since the min pixel intensity is 0 by default.